src/app/core/services/grouping/assignee-grouping-strategy.service.ts

```diff
@@ -13,30 +15,29 @@ import { GroupingStrategy } from './grouping-strategy.interface';
   providedIn: 'root'
 })
 export class AssigneeGroupingStrategy implements GroupingStrategy {
-  constructor(private githubService: GithubService) {}
+  constructor(private assigneeService: AssigneeService) {}
 
   /**
    * Retrieves data for a specific assignee.
-   * If it is the"No Assignee" group, unassigned issues are returned.
+   * If it is the "No Assignee" group, unassigned issues are returned.
    * Otherwise, issues assigned to the specified user are returned.
    */
-  getDataForGroup(issues: Issue[], key: GithubUser): Issue[] {
-    if (key === GithubUser.NO_ASSIGNEE) {
-      return this.getUnassignedData(issues);
-    }
-
-    return this.getDataAssignedToUser(issues, key);
+  getDataForGroup(issues: Issue[], key: Assignee): Issue[] {
+    return issues.filter(
+      (issue) => issue.assignees.includes(key.login) || (key === Assignee.WithoutAssignee && issue.assignees.length === 0)
+    );
   }
```

Comment:
In WATcher, the issue model can represent either a GitHub Issue or a GitHub Pull Request. Therefore, in getDataAssignedToUser, we use different filtering logic: for GitHub Issues, we retrieve those assigned to the target; for GitHub Pull Requests, we retrieve those created by the target.